chasm/visibility.go

```diff
@@ -57,10 +45,42 @@ func NewVisibility(
 			TransitionCount: 0,
 		},
 	}
-	visibility.GenerateTask(mutableContext)
+
+	visibility.generateTask(mutableContext)
 	return visibility
 }
 
+func NewVisibilityWithData(
+	mutableContext MutableContext,
+	searchAttributes map[string]*commonpb.Payload,
+	memo map[string]*commonpb.Payload,
```

Comment:
Maybe call this `userSearchAttributes` or `customSearchAttributes` and `userMemo`?